completed-components/exercise-4-ARIA/ProductHeader.tsx

```diff
@@ -20,36 +20,57 @@ const Banner = ({ shouldAnimate = false }: BannerProps) => {
 		}
 	};
 	const incrementSlide = () => {
-		if (currentSlideNum <= slideCount) {
+		if (currentSlideNum < slideCount) {
 			changeSlideNum(currentSlideNum + 1);
 		} else {
 			changeSlideNum(1);
 		}
 	};
+
+	const setInert = (node, slideNumber) => {
+		// workaround for React not recognizing inert
+		// https://github.com/facebook/react/pull/24730
+		return (
+			node && (currentSlideNum === slideNumber ? node.removeAttribute('inert', '') : node.setAttribute('inert', ''))
+		);
+	};
+
+	useEffect(() => {
+		updateSlidePercentage(((currentSlideNum - 1) / 3) * 100);
+	}, [currentSlideNum]);
+
+	async function animate() {
+		incrementSlide();
+	}
+
+	useEffect(() => {
+		if (shouldAnimate) animate();
+	}, []);
+
+	if (shouldAnimate) {
+		setTimeout(() => {
+			animate();
+		}, 3000);
+	}
+
 	return (
-		<div className="bg-black max-w-full w-full flex" id="banner" role="banner" aria-labelledby="carouselheading">
+		<div className="bg-black max-w-full w-full flex" id="banner" role="region" aria-labelledby="carouselheading">
 			<div className="flex max-w-[1400px] mx-auto">
 				<div className="mx-auto md:max-w-[65%] lg:max-w-[70%] flex">
-					<p id="carouselheading" className="sr-only">
+					<p id="carouselheading" className="hidden">
```

Comment:
Class name should not have switched from `sr-only` to `hidden`...not sure if this wasn't covered in the videos & that is how it was missed, or if I neglected to update alongside the video. As mentioned in my updated PR description, I'm having issues with git locally that I don't have time to troubleshoot & am going to leave this alone for now (will also be an issue for the other completed exercise files in this PR).